src/Workspaces/Core/Portable/Formatting/AbstractFormattingService.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    /// <summary>
+    /// Base implementation of C# and VB formatting services.
+    /// </summary>
+    internal abstract class AbstractFormattingService : IFormattingService
```

Comment:
Didn't we make it so that we could just export for multiple languages directly from here?